catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/monitor/email/SubscribedWorkspaceNotifier.java

```diff
@@ -0,0 +1,115 @@
+
+  private static EmailNotifier emailNotifierService;
+
+  private static FilterBuilder filterBuilder;
+
+  private static CatalogFramework catalogFramework;
+
+  private static final String WORKSPACE_TAG = "workspace";
+
+  private static final String QUERIES_ATTRIBUTE = "queries";
+
+  @SuppressWarnings("unused")
+  public void setSubjectIdentity(SubjectIdentity subjectIdentity) {
+    this.subjectIdentity = subjectIdentity;
+  }
+
+  @SuppressWarnings("unused")
+  public void setEmailNotifierService(EmailNotifier emailNotifierService) {
+    this.emailNotifierService = emailNotifierService;
+  }
+
+  @SuppressWarnings("unused")
+  public void setFilterBuilder(FilterBuilder filterBuilder) {
+    this.filterBuilder = filterBuilder;
+  }
+
+  @SuppressWarnings("unused")
+  public void setCatalogFramework(CatalogFramework catalogFramework) {
+    this.catalogFramework = catalogFramework;
+  }
+
+  @Override
+  public QueryResponse process(QueryResponse queryResponse) throws PluginExecutionException {
+    if (queryResponse == null) {
+      throw new PluginExecutionException("Cannot process null queryResponse");
+    }
+    if (!queryResponse.getResults().isEmpty()) {
+      String queryId = (String) queryResponse.getRequest().getProperties().get("id");
+      if (queryId != null) {
+        Metacard metacard = getWorkspaceMetacard(queryId);
+        WorkspaceMetacardImpl workspaceMetacard = WorkspaceMetacardImpl.from(metacard);
+        if (WorkspaceMetacardImpl.isWorkspaceMetacard(workspaceMetacard)) {
+          String queryIssuerEmail = subjectIdentity.getUniqueIdentifier(SecurityUtils.getSubject());
+          String workspaceOwnerEmail = AccessControlUtil.getOwner(workspaceMetacard);
+          if (workspaceOwnerEmail.equals(queryIssuerEmail)) {
+            emailNotifierService.sendEmailsForWorkspace(workspaceMetacard, 1L);
+          }
+        }
+      }
+    }
+    return queryResponse;
+  }
+
+  private Metacard getWorkspaceMetacard(String queryId) {
+    Filter filter = filterBuilder.allOf(getWorkspaceQueryFilters(queryId));
+
+    Query query = new QueryImpl(filter);
+    QueryRequest queryRequest = new QueryRequestImpl(query);
+
+    Metacard workspaceMetacard = null;
+    try {
+      QueryResponse queryResponse = catalogFramework.query(queryRequest);
+      workspaceMetacard =
+          queryResponse
+              .getResults()
+              .stream()
+              .map(Result::getMetacard)
+              .filter(Objects::nonNull)
+              .findFirst()
+              .orElse(null);
+
+    } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+      LOGGER.debug("Error querying for query id: {}.", queryId, e);
+    }
+    return workspaceMetacard;
+  }
+
+  private List<Filter> getWorkspaceQueryFilters(String queryId) {
+    List<Filter> filters = new ArrayList<>();
+    filters.add(filterBuilder.attribute(Metacard.TAGS).is().text(WORKSPACE_TAG));
```

Comment:
`Core.METACARD_TAGS`.  
Could also do `filters.allOf(filterBuilder.attribute(Metacard.TAGS).is().text(WORKSPACE_TAG), filters.add(filterBuilder.attribute(QUERIES_ATTRIBUTE).like().text(queryId)));`
`filterBuilder.attribute(QUERIES_ATTRIBUTE).is().text(queryId))`